src/os/Transaction.h

```diff
@@ -291,6 +291,11 @@ class Transaction {
   Transaction(const Transaction& other) = default;
   Transaction& operator=(const Transaction& other) = default;
 
+  void claim_and_reset() {
+    auto ret = Transaction();
+    std::swap(*this, ret);
```

Comment:
I'd rather we return ret here and avoid the instance being invalid in the first place.

```
Transaction claim_and_reset() {
  auto ret = Transaction();
  std::swap(*this, ret);
  return ret;
}
```